### ISSUE TITLE
Bug: unknown lvalue in section Service

### ISSUE THREAD

**Issue by Swallowtail23** [abraunegg/onedrive#1969]
### Describe the bug

CentOS 7, with 2.4.17 (I've tried both packaged release and also git clone). 

I'm assuming this is an issue in the service file, with EL7 perhaps not able to support those values?

The onedrive client appears to be running OK, I'm hoping these are cosmetic issues only in the service file.

```
May 23 20:40:37 emp75.simonandkate.lan systemd[1]: [/usr/lib/systemd/system/onedrive.service:12] Unknown lvalue 'ProtectHostname' in section 'Service'
May 23 20:40:37 emp75.simonandkate.lan systemd[1]: [/usr/lib/systemd/system/onedrive.service:14] Unknown lvalue 'ProtectKernelTunables' in section 'Service'
May 23 20:40:37 emp75.simonandkate.lan systemd[1]: [/usr/lib/systemd/system/onedrive.service:17] Unknown lvalue 'ProtectControlGroups' in section 'Service'
May 23 20:40:37 emp75.simonandkate.lan systemd[1]: [/usr/lib/systemd/system/onedrive.service:18] Unknown lvalue 'RestrictRealtime' in section 'Service'
```
This repeats a couple of times.

### Operating System Details

```shell
Linux emp75.simonandkate.lan 3.10.0-1160.66.1.el7.x86_64 #1 SMP Wed May 18 16:02:34 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

CentOS Linux release 7.9.2009 (Core)
```

### Client Installation Method

From Source

### OneDrive Account Type

Personal

### What is your OneDrive Application Version

onedrive v2.4.17-20-gddb9f1e

### What is your OneDrive Application Configuration

```shell
Configuration file successfully loaded
onedrive version                             = v2.4.17-20-gddb9f1e
Config path                                  = /root/.config/onedrive
Config file found in config path             = true
Config option 'sync_dir'                     = /root/OneDrive
Config option 'enable_logging'               = true
Config option 'log_dir'                      = /var/log/onedrive/
Config option 'disable_notifications'        = false
Config option 'min_notify_changes'           = 5
Config option 'skip_dir'                     =
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 5
Config option 'monitor_fullscan_frequency'   = 12
Config option 'dry_run'                      = false
Config option 'upload_only'                  = true
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = true
Config option 'check_nomount'                = true
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'application_id'               =
Config option 'azure_ad_endpoint'            =
Config option 'azure_tenant_id'              = common
Config option 'user_agent'                   =
Config option 'force_http_2'                 = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'sync_root_files'              = false
Selective sync 'sync_list' configured        = true
sync_list contents:
# Only back up the server's folder, ignore EVERYTHING else
EMP75-sync
Config option 'sync_business_shared_folders' = false
Business Shared Folders configured           = false
Config option 'webhook_enabled'              = false
```

### What is your 'curl' version

```shell
curl 7.29.0 (x86_64-redhat-linux-gnu) libcurl/7.29.0 NSS/3.53.1 zlib/1.2.7 libidn/1.28 libssh2/1.8.0
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp scp sftp smtp smtps telnet tftp
Features: AsynchDNS GSS-Negotiate IDN IPv6 Largefile NTLM NTLM_WB SSL libz unix-sockets
```

### Where is your 'sync_dir' located

Local

### What are all your system 'mount points'

```shell
sysfs on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
devtmpfs on /dev type devtmpfs (rw,nosuid,size=6057256k,nr_inodes=1514314,mode=755)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
tmpfs on /run type tmpfs (rw,nosuid,nodev,mode=755)
tmpfs on /sys/fs/cgroup type tmpfs (ro,nosuid,nodev,noexec,mode=755)
cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,release_agent=/usr/lib/systemd/systemd-cgroups-agent,name=systemd)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpuacct,cpu)
cgroup on /sys/fs/cgroup/pids type cgroup (rw,nosuid,nodev,noexec,relatime,pids)
cgroup on /sys/fs/cgroup/memory type cgroup (rw,nosuid,nodev,noexec,relatime,memory)
cgroup on /sys/fs/cgroup/net_cls,net_prio type cgroup (rw,nosuid,nodev,noexec,relatime,net_prio,net_cls)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)
cgroup on /sys/fs/cgroup/hugetlb type cgroup (rw,nosuid,nodev,noexec,relatime,hugetlb)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
configfs on /sys/kernel/config type configfs (rw,relatime)
/dev/mapper/cl-root on / type xfs (rw,relatime,attr2,inode64,noquota)
hugetlbfs on /dev/hugepages type hugetlbfs (rw,relatime)
debugfs on /sys/kernel/debug type debugfs (rw,relatime)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=35,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=11989)
mqueue on /dev/mqueue type mqueue (rw,relatime)
/dev/vda1 on /boot type xfs (rw,relatime,attr2,inode64,noquota)
/dev/vdd1 on /mnt/4tbR5 type xfs (rw,relatime,attr2,inode64,noquota)
/dev/vdb1 on /mnt/data type ext4 (rw,relatime,data=ordered)
/dev/vdb1 on /home type ext4 (rw,relatime,data=ordered)
/dev/vdb1 on /restore type ext4 (rw,relatime,data=ordered)
/dev/vdc1 on /mnt/8gb type ext4 (rw,relatime,data=ordered)
/dev/vdc1 on /backup type ext4 (rw,relatime,data=ordered)
/dev/vdc1 on /backup2 type ext4 (rw,relatime,data=ordered)
sunrpc on /var/lib/nfs/rpc_pipefs type rpc_pipefs (rw,relatime)
tmpfs on /run/user/0 type tmpfs (rw,nosuid,nodev,relatime,size=1213768k,mode=700)
binfmt_misc on /proc/sys/fs/binfmt_misc type binfmt_misc (rw,relatime)
```

### What are all your local file system partition types

```shell
NAME        FSTYPE      LABEL UUID                                   MOUNTPOINT
vda
├─vda1      xfs               c12a9623-1765-4a7a-9906-a549587d3dba   /boot
└─vda2      LVM2_member       OmB0uN-G6Xd-AQyT-1M5L-bujH-CS3Q-V3B2Fz
  ├─cl-root xfs               26c9b195-794e-4254-91df-8b602734e046   /
  └─cl-swap swap              b032ae64-e7d5-4ba2-9a53-f6956ff326a8   [SWAP]
vdb
└─vdb1      ext4              e4d6a689-6818-4386-bc0f-b6e473b8b33a   /mnt/data
vdc
└─vdc1      ext4              31e12b06-7766-4788-a638-418ccaaf2e6f   /mnt/8gb
vdd
└─vdd1      xfs               91725d5b-820b-4f2a-a73f-ac92d9920eae   /mnt/4tbR5
```

### How do you use 'onedrive'

Yes, multiple devices use same account with the onedrive client, including from RHEL8. 

### Steps to reproduce the behaviour

Start the client :)

### Complete Verbose Log Output

```shell
Not a crash, and see above for log entry
```

**Comment by abraunegg** [comment]
@Swallowtail23 
Unfortunately this is not a bug with this client.

The issue is with CentOS 7 and the version of systemd it is using.

Your only option is to comment out the items in the systemd service file to resolve this issue to prevent those messages from occurring.

The alternative is to log a bug here https://bugzilla.redhat.com/ for systemd ... as this is a systemd issue.

**Comment by Swallowtail23** [comment]
All good, and thank you. At least now it's in here in case anyone else runs across it - I searched for a while before finding the answer; I've commented out the lines and yes it resolves it. 
No point logging a bug at RH - the systemd version on RHEL7/CentOS7 does not support those flags in the service file, and they won't do a major rev update in 7.
Perhaps a comment in your doco that the service file installed will trigger this on RH/CentOS versions < 8?
